functions/AuthUtils.js

```diff
@@ -0,0 +1,186 @@
+            },
+        },
+    };
+    return await jwt.sign(netlifyTokenData, process.env.TOKEN_SECRET);
+};
+
+const generateAuth0LoginCookie = (nonce, encodedStateStr) => {
+    const cookieData = { nonce, state: encodedStateStr };
+    return cookie.serialize('auth0_login_cookie', JSON.stringify(cookieData), {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        path: '/',
+        maxAge: 30 * 60 * 1000,
+        httpOnly: true,
+    });
+};
+
+const generateEncodedStateString = (route) => {
+    const state = { route: route || '/', nonce: generators.nonce() };
+    const stateBuffer = Buffer.from(JSON.stringify(state));
+    return stateBuffer.toString('base64');
+};
+
+const generateAuth0LoginResetCookie = () => {
+    return cookie.serialize('auth0_login_cookie', 'Auth0 Login Cookie Reset', {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        httpOnly: true,
+        path: '/',
+        maxAge: new Date(0),
+    });
+};
+
+const generateLogoutCookie = () => {
+    return cookie.serialize('nf_jwt', 'Logout Cookie', {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        path: '/',
+        maxAge: new Date(0),
+        httpOnly: true,
+    });
+};
+
+const generateNetlifyCookieFromAuth0Token = async (tokenData) => {
+    const netlifyToken = await generateNetlifyJWT(tokenData);
+
+    const twoWeeks = 14 * 24 * 3600000;
+    return cookie.serialize('nf_jwt', netlifyToken, {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        path: '/',
+        maxAge: twoWeeks,
+    });
+};
+
+const getCallbackParams = (openIDClient, event) => {
+    /* NOTE: method, body, and url are all required for the openIDClient to work with
+    the request*/
+    const req = {
+        method: 'POST',
+        body: event.body,
+        url: event.headers.host,
+    };
+    //callbackParams documentation - https://github.com/panva/node-openid-client/tree/master/docs#clientcallbackparamsinput
+    return openIDClient.callbackParams(req);
+};
+
+const generateAuth0LogoutUrl = () => {
+    const auth0DomainLogout = `https://${process.env.AUTH0_DOMAIN}/v2/logout`;
+    const urlReturnTo = `returnTo=${encodeURIComponent(process.env.URL)}`;
+    const urlClientId = `client_id=${process.env.AUTH0_CLIENT_ID}`;
+    return `${auth0DomainLogout}?${urlReturnTo}&${urlClientId}`;
+};
+
+const handleLogin = async (event) => {
+    if (!event || !event.headers) {
+        throw new Error('Malformed event');
+    }
+    const openIDClient = await getOpenIDClient();
+    const referer = event.headers.referer;
+    const encodedStateStr = generateEncodedStateString(referer);
+    const nonce = generators.nonce();
+
+    //authorizationUrl docs - https://github.com/panva/node-openid-client/tree/master/docs#clientauthorizationurlparameters
+    const authRedirectURL = openIDClient.authorizationUrl({
+        scope: 'openid email profile',
+        response_mode: 'form_post',
+        nonce,
+        state: encodedStateStr,
```

Comment:
```suggestion
        state: generateEncodedStateString(referer),
```

Reply:
Just relized I need encodedStateStr in a variable because they are used again in the generateAuth0LoginCookie call